plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -153,12 +151,14 @@ def run(self):
             return report
 
         except Exception, e:
-            _logger.exception('sync failed')
             for step, value in self.progress_status.iteritems():
                 if value.get('state') == constants.STATE_RUNNING:
                     value['state'] = constants.STATE_FAILED
                     value['error'] = str(e)
             self.set_progress()
+            # this is the new preferred way
+            if isinstance(e, PulpCodedException):
```

Comment:
Consider moving this codepath to its own except block above `except Exception`. It would be:

```
try:
    stuff
except PulpCodedException
    raise
except Exception, e:
    existing code before the addition in this PR
```